internal/pkg/cli/job_init.go

```diff
@@ -0,0 +1,297 @@
+Deployed resources (such as your job, logs) will contain this job's name and be tagged with it.`
+
+	jobInitDockerfileHelpPrompt = "Dockerfile to use for building your job's container image."
+)
+
+// const (
+// 	fmtAddJobToAppStart    = "Creating ECR repositories for job %s."
+// 	fmtAddJobToAppFailed   = "Failed to create ECR repositories for job %s.\n"
+// 	fmtAddJobToAppComplete = "Created ECR repositories for job %s.\n"
+// )
+
+type initJobVars struct {
+	*GlobalOpts
+	Name           string
+	DockerfilePath string
+}
+
+type initJobOpts struct {
+	initJobVars
+
+	// Interfaces to interact with dependencies.
+	fs          afero.Fs
+	ws          svcManifestWriter
+	store       store
+	appDeployer appDeployer
+	prog        progress
+
+	// Outputs stored on successful actions.
+	manifestPath string
+}
+
+func newInitJobOpts(vars initJobVars) (*initJobOpts, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("couldn't connect to config store: %w", err)
+	}
+
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, fmt.Errorf("workspace cannot be created: %w", err)
+	}
+
+	p := sessions.NewProvider()
+	sess, err := p.Default()
+	if err != nil {
+		return nil, err
+	}
+
+	return &initJobOpts{
+		initJobVars: vars,
+
+		fs:          &afero.Afero{Fs: afero.NewOsFs()},
+		store:       store,
+		ws:          ws,
+		appDeployer: cloudformation.New(sess),
+		prog:        termprogress.NewSpinner(),
+	}, nil
+}
+
+// Validate returns an error if the flag values passed by the user are invalid.
+func (o *initJobOpts) Validate() error {
+	if o.AppName() == "" {
+		return errNoAppInWorkspace
+	}
+	if o.Name != "" {
+		if err := validateSvcName(o.Name); err != nil {
+			return err
+		}
+	}
+	if o.DockerfilePath != "" {
+		if _, err := o.fs.Stat(o.DockerfilePath); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Ask prompts for fields that are required but not passed in.
+func (o *initJobOpts) Ask() error {
+	if err := o.askJobName(); err != nil {
+		return err
+	}
+	if err := o.askDockerfile(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Execute writes the service's manifest file and stores the service in SSM.
+func (o *initJobOpts) Execute() error {
+	app, err := o.store.GetApplication(o.AppName())
+	if err != nil {
+		return fmt.Errorf("get application %s: %w", o.AppName(), err)
+	}
+
+	manifestPath, err := o.createManifest()
+	if err != nil {
+		return err
+	}
+	o.manifestPath = manifestPath
+
+	o.prog.Start(fmt.Sprintf(fmtAddSvcToAppStart, o.Name))
+	if err := o.appDeployer.AddServiceToApp(app, o.Name); err != nil {
+		o.prog.Stop(log.Serrorf(fmtAddSvcToAppFailed, o.Name))
+		return fmt.Errorf("add job %s to application %s: %w", o.Name, o.AppName(), err)
+	}
+	o.prog.Stop(log.Ssuccessf(fmtAddSvcToAppComplete, o.Name))
+
+	if err := o.store.CreateService(&config.Service{
+		App:  o.AppName(),
+		Name: o.Name,
+		Type: "Scheduled Job",
+	}); err != nil {
```

Comment:
Similarly: Do we need to rename the service entity to `Workload` in the `config` package?